src/wp-includes/media.php

```diff
@@ -5564,6 +5686,123 @@ function wp_get_loading_attr_default( $context ) {
 	return 'lazy';
 }
 
+/**
+ * Gets loading optimization attributes.
+ *
+ * @since 6.3.0
+ *
+ * @param string  $tag_name The tag name.
+ * @param array[] $attr     Array of the attributes for the tag.
+ * @param string  $context  Context for the element for which the loading optimization attribute is requested.
+ *
+ * @return array[] Loading optimization attributes.
+ */
+function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
+	global $wp_query;
+
+	$loading_attrs = array();
+
+	// For now this function only supports images.
+	if ( 'img' !== $tag_name ) {
+		return $loading_attrs;
+	}
+
+	// For any resources, width and height must be provided, to avoid layout shifts.
+	if ( ! isset( $attr['width'] ) || ! isset( $attr['height'] ) ) {
+		return $loading_attrs;
+	}
+
+	if ( isset( $attr['loading'] ) && 'lazy' === $attr['loading'] ) {
+		if ( isset( $attr['fetchpriority'] ) && 'high' === $attr['fetchpriority']
```

Comment:
The way you check those conditions here is not quite correct:
* We need to enter the general condition as soon as a custom `loading` attribute is provided - _regardless_ of its value.
* The fact that someone provided `lazy` as the value is only important for the `_doing_it_wrong()` warning.
```suggestion
	if ( isset( $attr['loading'] ) ) {
		if ( 'lazy' === $attr['loading'] && isset( $attr['fetchpriority'] ) && 'high' === $attr['fetchpriority']
```

Reply:
Modified.